src/main/java/de/teamlapen/vampirism/entity/vampire/EntityBasicVampire.java

```diff
@@ -282,35 +280,37 @@ protected void updateEntityAttributes() {
     }
 
     /**
-     * sets the possible action sets and returns a random action set with setAvailableActions() of {@linkplain EntityVampirism}
+     * returns the available actions based on {@link EntityClass}
      * 
      * @returns an action set
      */
-    protected List<IEntityAction> getAvailableActions(int level) {
-        List<IEntityAction[]> actionstmp = new ArrayList<IEntityAction[]>();
-        switch (level) {
-            case 0:
-                actionstmp.add(new IEntityAction[] { EntityActions.entity_regeneration });
-                actionstmp.add(new IEntityAction[] { EntityActions.entity_speed });
-                actionstmp.add(new IEntityAction[] { EntityActions.entity_bat_spawn, EntityActions.entity_regeneration });
-                actionstmp.add(new IEntityAction[] { EntityActions.entity_bat_spawn });
+    @Override
```

Comment:
It would be perfect if this could somehow be moved into the action registry and the actions themselves.
Maybe each action is registered with (or provides) the entity class(es) it fits as well as some kind of indication if it is rather a lower or a higher tier action. 
Then the code within the entity itself would only have to provide some sort of level (basic vampire: low to medium, advanced vampire: high, baron: ultimate) and it's class